src/app/routines/Routines.tsx

```diff
@@ -29,14 +31,30 @@ const Routines = () => {
     navigate(AppRoutes.ROUTINE_CREATE);
   };
 
+  const debouncedHandleSearch = debounce((event: React.ChangeEvent<HTMLInputElement>) => {
+    if (tabValue === 'my-routines') {
```

Comment:
minor: would be better to use constant of that string, cause we are using it in 2 places and easy to do typo